ios/RN/RNCamera.m

```diff
@@ -1437,7 +1436,10 @@ - (void)sessionWasInterrupted:(NSNotification *)notification
         // if we have audio, stop it so preview resumes
         // it will eventually be re-loaded the next time recording
         // is requested, although it will flicker.
-        [self removeAudioCaptureSessionInput];
+        dispatch_async(self.sessionQueue, ^{
```

Comment:
why this fix the issue?

Reply:
We remove the audio input when the session is interrupted so the session can resume, even while audio is not available.  Removing the audio input automatically starts/resumes the session. However, when this was done in the same session interrupted event, some kind of race condition was messing up the internal session `isRunning` flag. Apple states that this value is not thread safe, so it makes sense. Moving the call to the session queue removed the concurrency issues and the flag is now properly updated.

Reply:
tks for the explanations